types/stylelint/index.d.ts

```diff
@@ -88,31 +88,155 @@ declare namespace stylelint {
 	 * Configuration.
 	 */
 	export type Config = {
+		/**
+		 * Allows to extend an existing configuration. Configurations can bundle plugins, custom syntaxes,
+		 * options, and configure rules. They can also extend other configurations
+		 *
+		 * @see [extends](https://stylelint.io/user-guide/configure/#extends)
+		 */
 		extends?: ConfigExtends;
+		/**
+		 * Custom rules or sets of custom rules built to support methodologies, toolsets,
+		 * non-standard CSS features, or very specific use cases
+		 *
+		 * @see [plugins](https://stylelint.io/user-guide/configure/#plugins)
+		 */
 		plugins?: ConfigPlugins;
+		/**
+		 * Plugin functions
+		 *
+		 * @see [The anatomy of a plugin](https://stylelint.io/developer-guide/plugins/#the-anatomy-of-a-plugin)
+		 */
 		pluginFunctions?: {
 			[pluginName: string]: Rule;
 		};
+		/**
+		 * A glob or array of globs to ignore specific files
+		 *
+		 * @default 'node_modules'
+		 *
+		 * @see [ignorefiles](https://stylelint.io/user-guide/configure/#ignorefiles)
+		 */
 		ignoreFiles?: ConfigIgnoreFiles;
+		/**
+		 * Pattern of files to ignore
+		 */
 		ignorePatterns?: string;
+		/**
+		 * An object containing the configured rules
+		 *
+		 * @see [rules](https://stylelint.io/user-guide/configure/#rules)
+		 */
 		rules?: ConfigRules;
+		/**
+		 * Only register problems for rules with an "error"-level severity (ignore "warning"-level)
+		 *
+		 * @see [quiet](https://stylelint.io/user-guide/options/#quiet)
+		 */
 		quiet?: boolean;
+		/**
+		 * A string to specify the name of a formatter or a path to a custom formatter function
+		 *
+		 * @see [formatter](https://stylelint.io/user-guide/configure#formatter)
+		 */
 		formatter?: FormatterType | Formatter;
+		/**
+		 * A string to set the default severity level for all rules that do not have a severity
+		 * specified in their secondary options
+		 *
+		 * @see [defaultseverity](https://stylelint.io/user-guide/configure#defaultseverity)
+		 */
 		defaultSeverity?: Severity;
+		/**
+		 * A boolean value indicating if 'stylelint-disable' comments will be ignored
+		 *
+		 * @see [ignoredisables](https://stylelint.io/user-guide/configure#ignoredisables)
+		 */
 		ignoreDisables?: boolean;
+		/**
+		 * Report 'stylelint-disable' comments that don't match any lints that need to be disabled
+		 *
+		 * @see [reportneedlessdisables](https://stylelint.io/user-guide/configure#reportneedlessdisables)
+		 */
 		reportNeedlessDisables?: DisableSettings;
+		/**
+		 * Report 'stylelint-disable' comments that don't match rules that are specified in the configuration object
+		 *
+		 * @see [reportinvalidscopedisables](https://stylelint.io/user-guide/configure#reportinvalidscopedisables)
+		 */
 		reportInvalidScopeDisables?: DisableSettings;
+		/**
+		 * Report 'stylelint-disable' comments without a description
```

Comment:
minor

Like l175, you should replace `'stylelint-disable'` with `configuration` for l169 l163 l157.
see https://stylelint.io/user-guide/configure/#configurationcomment